### ISSUE TITLE
iperf test completed successfully but got a return code error "Could not fetch resource..." during cleaning up benchmark.

### ISSUE THREAD

**Issue by hungvelo** [GoogleCloudPlatform/PerfKitBenchmarker#168]
2015-03-13 16:20:39,820 156efb06 MainThread iperf(1/1) pkb.py:217 INFO     Cleaning up benchmark iperf
2015-03-13 16:20:39,821 156efb06 MainThread iperf(1/1) vm_util.py:339 INFO     Running: /usr/bin/ssh -A -p 22 perfkit@146.148.58.238 -2 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o IdentitiesOnly=yes -o PreferredAuthentications=publickey -o PasswordAuthentication=no -o ConnectTimeout=5 -o GSSAPIAuthentication=no -o ServerAliveInterval=30 -o ServerAliveCountMax=10 -i /tmp/perfkitbenchmarker/run_156efb06/perfkitbenchmarker_keyfile pkill -9 iperf
2015-03-13 16:20:40,707 156efb06 MainThread iperf(1/1) vm_util.py:353 DEBUG    Ran /usr/bin/ssh -A -p 22 perfkit@146.148.58.238 -2 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o IdentitiesOnly=yes -o PreferredAuthentications=publickey -o PasswordAuthentication=no -o ConnectTimeout=5 -o GSSAPIAuthentication=no -o ServerAliveInterval=30 -o ServerAliveCountMax=10 -i /tmp/perfkitbenchmarker/run_156efb06/perfkitbenchmarker_keyfile pkill -9 iperf. Got return code (0). STDOUT: STDERR: Warning: Permanently added '146.148.58.238' (ECDSA) to the list of known hosts.^M

2015-03-13 16:20:40,708 156efb06 Thread-4 iperf(1/1) vm_util.py:339 INFO     Running: gcloud compute instances delete perfkit-156efb06-1 --project elite-firefly-786 --format json --quiet --zone us-central1-a
2015-03-13 16:20:40,709 156efb06 Thread-5 iperf(1/1) vm_util.py:339 INFO     Running: gcloud compute instances delete perfkit-156efb06-0 --project elite-firefly-786 --format json --quiet --zone us-central1-a
2015-03-13 16:21:08,083 156efb06 Thread-4 iperf(1/1) vm_util.py:353 DEBUG    Ran gcloud compute instances delete perfkit-156efb06-1 --project elite-firefly-786 --format json --quiet --zone us-central1-a. Got return code (0). STDOUT: []
STDERR: Deleted [https://www.googleapis.com/compute/v1/projects/elite-firefly-786/zones/us-central1-a/instances/perfkit-156efb06-1].

2015-03-13 16:21:08,083 156efb06 Thread-4 iperf(1/1) vm_util.py:339 INFO     Running: gcloud compute instances describe perfkit-156efb06-1 --project elite-firefly-786 --format json --quiet --zone us-central1-a
2015-03-13 16:21:08,847 156efb06 Thread-4 iperf(1/1) vm_util.py:353 DEBUG    Ran gcloud compute instances describe perfkit-156efb06-1 --project elite-firefly-786 --format json --quiet --zone us-central1-a. Got return code (1). STDOUT: STDERR: ERROR: (gcloud.compute.instances.describe) Could not fetch resource:
 - The resource 'projects/elite-firefly-786/zones/us-central1-a/instances/perfkit-156efb06-1' was not found

**Comment by cmccoy** [comment]
Did the process exit successfully?  We [run](https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/resource.py#L101-L108) `instances describe` after `instances delete` to verify that the delete succeeded.  The command is expected to fail whenever the delete succeeded.

**Comment by hungvelo** [comment]
Yes, the process exited successfully. As you could see from the log and also as you described, we ran instance delete perfkit-156efb06-1 followed by instance describe perfkit-156efb06-1, and that is expected to fail. But if that is expected, we shouldn't generate a false alarm with an ERROR, right? Just my .02 cents :-). Thank you.

**Comment by cmccoy** [comment]
Agreed - this is something we'd like to improve.

Generally these "expected" errors aren't printed to stderr, just the log file at a low level (here, DEBUG).  The "ERROR" is actually from the stderr of `gcloud`, and we *do* want to log the output of every command to the log file for debugging.
